floyd/cli/utils.py

```diff
@@ -55,28 +60,118 @@ def get_data_id(data_str):
         return data_str
 
 
-def normalize_data_name(data_name):
-    if data_name.endswith('/output'):
-        name_parts = data_name.split('/')
-        if len(name_parts) <= 4:
-            name_parts.insert(1, 'projects')
-            data_name = '/'.join(name_parts)
-        return data_name
+def normalize_data_name(raw_name, default_username=None, default_dataset_name=None):
+    if raw_name.endswith('/output'):
+        return normalize_job_name(raw_name[:-len('/output')], default_username, default_dataset_name) + '/output'
+
+    name_parts = raw_name.split('/')
+
+    username = default_username or current_username()
+    name = default_dataset_name or current_experiment_name()
```

Comment:
Not sure if we should use `current_experiment_name` when looking for a dataset name. Should this be `current_data_name`?